### PR TITLE
Split the unconditional runtime dependencies into extras

### PR DESCRIPTION
**Pull request by jessicamack**
All the runtime dependencies of `awx-plugins-core` are unconditional now. To take into account being able to depend only on certain plugins, I've split them into optional ones.

### CODE REVIEW

**Comment by webknjaz** [comment]
@jessicamack to fix the CI, change `pyproject.toml` to `pyproject.toml --all-extras` on these lines: https://github.com/ansible/awx-plugins/blob/c7fc0a1/.github/workflows/pip-tools.yml#L102-L128. Also, here https://github.com/ansible/awx-plugins/blob/c7fc0a1/tox.ini#L15, after `deps =`, add `extras =` with a column containing all the extras, one per line, use two-space indents.

File: pyproject.toml
Comment (webknjaz) [inline pyproject.toml]: Keep the empty list declared instead of commenting this out.

**Comment by webknjaz** [comment]
@jessicamack have you seen https://github.com/ansible/awx-plugins/pull/52#issuecomment-2445712910?

**Review by djyasin — APPROVED** [review]

**Comment by webknjaz** [comment]
UPD: this is failing because of a bug in tox (https://github.com/tox-dev/tox/issues/3433) which is blocking the PR.

File: pyproject.toml
Comment (webknjaz) [inline pyproject.toml]: Example:
```suggestion
credentials-hashivault-kv = [
```

File: tox.ini
Comment (webknjaz) [inline tox.ini]: Example:
```suggestion
  credentials-hashivault-kv
```

File: tox.ini
Comment (webknjaz) [inline tox.ini]: revert
```suggestion
```

File: tox.ini
Comment (webknjaz) [inline tox.ini]: revert
```suggestion
```

File: tox.ini
Comment (webknjaz) [inline tox.ini]: revert
```suggestion
```

File: tox.ini
Comment (webknjaz) [inline tox.ini]: revert
```suggestion
```

File: tox.ini
Comment (webknjaz) [inline tox.ini]: revert
```suggestion
```

**Review by webknjaz — DISMISSED** [review]
@jessicamack so I learned that [PEP 685](https://packaging.python.org/en/latest/specifications/core-metadata/#provides-extra-multiple-use) will make the format of extras stricter in the internal core packaging metadata version 2.3. The user-facing extra names might still be allowed to be non-normalized. But in general, it's best to just follow that format to avoid confusion. Evidently, some tools are going to get confused and hit corner cases if we don't do this.

Action items:
* [x] Change extra names in `pyproject.toml` to have `-` instead of `_`
* [x] Change extra names in `tox.ini` to have `-` instead of `_`

**Comment by webknjaz** [comment]
@chrismeyersfsu could you confirm that the extra names look fine? Not that they can't be changed, but it's a chance to update them if there's anything obvious that stands out.

**Review by webknjaz — APPROVED** [review]
Waiting from Chris' take on the name, and then we can merge.

**Comment by chrismeyersfsu** [comment]
So I understand, after this merges, awx will NOT get these extra modules from awx-plugins.interfaces unless awx adds the optional directive I.e. awx-plugins[x,y,z]

**Comment by webknjaz** [comment]
Yes. Although, it could just start depending on the version you released on Monday until things settle and they those extras that remain post split activities would need to be listed along with the version bump.

**Review by chrismeyersfsu — APPROVED** [review]

**Comment by webknjaz** [comment]
Now a companion dep update is needed in awx.

**Pull request merged**
